src/sentry/tasks/data_export.py

```diff
@@ -71,86 +76,52 @@ def assemble_download(data_export_id):
         return data_export.email_failure(message="Internal processing failure")
 
 
-def process_issue_by_tag(data_export, file, limit=None):
+def process_billing_report(data_export, file):
+    # TODO(Leander): Implement processing for Billing Reports
+    raise NotImplementedError("Billing report processing has not been implemented yet")
+
+
+def process_issues_by_tag(data_export, file, limit, environment_id):
     """
     Convert the tag query to a CSV, writing it to the provided file.
     Returns the suggested file name.
-    (Adapted from 'src/sentry/web/frontend/group_tag_export.py')
     """
-    # Get the pertaining project
+    payload = data_export.query_info
     try:
-        payload = data_export.query_info
-        project = Project.objects.get(id=payload["project_id"])
-    except Project.DoesNotExist as error:
-        metrics.incr("dataexport.error", instance=six.text_type(error))
-        logger.error("dataexport.error: {}".format(six.text_type(error)))
-        raise DataExportError("Requested project does not exist")
-
-    # Get the pertaining issue
-    try:
-        group, _ = get_group_with_redirect(
-            payload["group_id"], queryset=Group.objects.filter(project=project)
+        processor = IssuesByTagProcessor(
+            project_id=payload["project_id"],
+            group_id=payload["group_id"],
+            key=payload["key"],
+            environment_id=environment_id,
         )
-    except Group.DoesNotExist as error:
+    except ProcessingError as error:
         metrics.incr("dataexport.error", instance=six.text_type(error))
         logger.error("dataexport.error: {}".format(six.text_type(error)))
-        raise DataExportError("Requested issue does not exist")
-
-    # Get the pertaining key
-    key = payload["key"]
-    lookup_key = six.text_type("sentry:{}").format(key) if tagstore.is_reserved_key(key) else key
-
-    # If the key is the 'user' tag, attach the event user
-    def attach_eventuser(items):
-        users = EventUser.for_tags(group.project_id, [i.value for i in items])
-        for item in items:
-            item._eventuser = users.get(item.value)
-
-    # Create the fields/callback lists
-    if key == "user":
-        callbacks = [attach_eventuser]
-        fields = [
-            "value",
-            "id",
-            "email",
-            "username",
-            "ip_address",
-            "times_seen",
-            "last_seen",
-            "first_seen",
-        ]
-    else:
-        callbacks = []
-        fields = ["value", "times_seen", "last_seen", "first_seen"]
-
-    # Example file name: Issues-by-Tag-project10-user__721.csv
-    file_details = six.text_type("{}-{}__{}").format(project.slug, key, data_export.id)
-    file_name = get_file_name(ExportQueryType.ISSUES_BY_TAG_STR, file_details)
+        raise DataExportError(error)
```

Comment:
This one block essentially runs all of the 'processing code' through the IssuesByTag constructor, skipping all the way the way to writing the CSV!